test/server/MongoDriver.test.ts

```diff
@@ -0,0 +1,207 @@
+    })
+
+    test('aggregate only returns one object', async () => {
+      const findByIdSpy = jest.spyOn(CategorySchema, 'findById')
+      const mockAggregate = (CategorySchema.aggregate = jest
+        .fn()
+        .mockImplementation(async () => [
+          categoryFixture,
+          { ...categoryFixture, _id: '123' },
+        ]))
+
+      const category = await MongoDriver.getEntity(
+        CategorySchema,
+        mockObjectId,
+        [{ $match: { _id: mockObjectId } }, { $limit: 1 }]
+      )
+
+      expect(findByIdSpy).toHaveBeenCalledTimes(0)
+      expect(mockAggregate).toHaveBeenCalledTimes(1)
+      expect(category).toEqual(categoryFixture)
+    })
+    test('valid call returns correct data', async () => {
+      const mockFindById = (CategorySchema.findById = jest
+        .fn()
+        .mockImplementation(async () => categoryFixture))
+
+      const category = await MongoDriver.getEntity(CategorySchema, mockObjectId)
+
+      expect(mockFindById).toHaveBeenCalledTimes(1)
+      expect(mockFindById).lastCalledWith(mockObjectId)
+      expect(category).toEqual(categoryFixture)
+    })
+
+    test('entity not found returns 404', async () => {
+      const mockFindById = (CategorySchema.findById = jest
+        .fn()
+        .mockImplementation(async () => undefined))
+
+      try {
+        await MongoDriver.getEntity(CategorySchema, mockObjectId)
+      } catch (error) {
+        expect(mockFindById).toHaveBeenCalledTimes(1)
+        expect(error).toBeInstanceOf(ApiError)
+        expect(error.message).toBe('Entity does not exist')
+        expect(error.statusCode).toBe(404)
+      }
+    })
+  })
+
+  describe('getEntities', () => {
+    test('uses aggregate if defined', async () => {
+      const findByIdSpy = jest.spyOn(CategorySchema, 'find')
+      const mockAggregate = (CategorySchema.aggregate = jest
+        .fn()
+        .mockImplementation(async () => [categoryFixture]))
+
+      const category = await MongoDriver.getEntities(CategorySchema, [
+        { $match: { _id: mockObjectId } },
+        { $limit: 1 },
+      ])
+
+      expect(findByIdSpy).toHaveBeenCalledTimes(0)
+      expect(mockAggregate).toHaveBeenCalledTimes(1)
+      expect(category).toEqual([categoryFixture])
+    })
+    test('valid call returns correct data', async () => {
+      const mockFind = (CategorySchema.find = jest
+        .fn()
+        .mockImplementation(async () => [categoryFixture]))
+
+      const categories = await MongoDriver.getEntities(CategorySchema)
+
+      expect(mockFind).toHaveBeenCalledTimes(1)
+      expect(categories).toEqual([categoryFixture])
+    })
+  })
+
+  describe('createEntity', () => {
+    test('returns category', async () => {
+      const mockCreate = (CategorySchema.create = jest
+        .fn()
+        .mockImplementation(async () => categoryFixture))
+
+      const category = await MongoDriver.createEntity(CategorySchema, {
+        ...categoryFixture,
+        _id: undefined,
```

Comment:
should this be not undefined?

Reply:
when creating an entity the id wont be known, just testing here with data it will have.

Reply:
Ah, right.